compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -700,7 +700,7 @@ object PatternMatcher {
           }
           val newArgs =
             for {
-              (rhs, actual) <- seenVars.toList
+              (rhs, actual) <- seenVars.toList.sortBy(_._2.name.toString)
```

Comment:
We have an ordering on `Name` (i.e. `Names.NameOrdering`), you shouldn't need to do `toString`

Reply:
This is just to test if this is the source of the idempotency failure.